### DISCUSSION TITLE
Add starting radius to circle and sector

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [Turfjs/turf#2778] [Feature proposals]
Having Circle and Sector accept an option like "initRadius" , where the circle or sector will start after that radius.

This can be easily done internally with the `Misc.mask` 

Why to achieve something like a torus for circles and sectors really easily and conveniently

**Comment by LamprosPitsillos** [comment]
```js
var radius = 5;
var bearing1 = 25;
var bearing2 = 45;


var A_center = turf.point([-75, 40]);
var B_center = turf.point([-75.02, 40.01]);

var A_sector = turf.sector(A_center, radius, bearing1, bearing2);
var A_sector_mask = turf.sector(A_center, radius-2, bearing1, bearing2);


var x = turf.difference(turf.featureCollection([A_sector, A_sector_mask]));

var B_sector = turf.sector(B_center, radius, bearing1, bearing2);
var B_sector_mask = turf.sector(B_center, radius-2, bearing1, bearing2);

const y = turf.mask(B_sector, B_sector_mask);
return turf.featureCollection([x,y]);
```

![image](https://github.com/user-attachments/assets/3e8928ac-16d5-4a49-823f-3193aa724cb8)

Turns out i cant achieve what i want using mask ( documented behavior ) "Returns Masked Polygon (**exterior ring** with holes)"
Nor using the turf difference, am i doing something wrong?

**Reply by smallsaucepan** [reply]
Hi @LamprosPitsillos. Is your issue the narrow sections running back to the center of your circle? If so, that might be because (due to some precision issues) your sector mask is _slightly_ narrower than your sector when you call difference.

Instead, I would suggest making your mask the full circle. That way it will correctly chop off those lines as well:

```typescript
// var A_sector_mask = turf.sector(A_center, radius-2, bearing1, bearing2);
var A_circle_mask = turf.circle(A_center, radius-2);

// var x = turf.difference(turf.featureCollection([A_sector, A_sector_mask]));
var x = turf.difference(turf.featureCollection([A_sector, A_circle_mask]));
```

Let us know if that helps!
